src/Middleware/Rewrite/src/RedirectToWwwRuleBase.cs

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Extensions;
+using Microsoft.AspNetCore.Rewrite.Logging;
+using Microsoft.Net.Http.Headers;
+
+namespace Microsoft.AspNetCore.Rewrite
+{
+    internal abstract class RedirectToWwwRuleBase : IRule
+    {
+        private const string WwwDot = "www.";
+        private const string Localhost = "localhost";
+
+        public readonly int _statusCode;
+        public readonly string[] _domains;
+
+        protected RedirectToWwwRuleBase(int statusCode)
+        {
+            _statusCode = statusCode;
+        }
+
+        protected abstract bool RedirectToWww { get; }
+
+        protected RedirectToWwwRuleBase(int statusCode, params string[] domains)
+        {
+            if (domains == null)
+            {
+                throw new ArgumentNullException(nameof(domains));
+            }
+
+            if (domains.Length < 1)
+            {
+                throw new ArgumentException($"One or more {nameof(domains)} must be provided.");
+            }
+
+            _domains = domains;
+            _statusCode = statusCode;
+        }
+
+        public virtual void ApplyRule(RewriteContext context)
+        {
+            var req = context.HttpContext.Request;
+
+            if (req.Host.Host.Equals(Localhost, StringComparison.OrdinalIgnoreCase))
+            {
+                context.Result = RuleResult.ContinueRules;
+                return;
+            }
+
+            if (RedirectToWww && req.Host.Value.StartsWith(WwwDot, StringComparison.OrdinalIgnoreCase))
+            {
+                context.Result = RuleResult.ContinueRules;
+                return;
+            }
+
+            if (!RedirectToWww && !req.Host.Value.StartsWith(WwwDot, StringComparison.OrdinalIgnoreCase))
+            {
+                context.Result = RuleResult.ContinueRules;
+                return;
+            }
+
+            if (_domains != null)
```

Comment:
This null check can be removed since there's a null check in ctor.

Reply:
It could be null via the `protected RedirectToWwwRuleBase(int statusCode)` ctor.

Reply:
You're right, my bad.